01_fib/nicholascross/main.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
```

Comment:
I agree, it is potentially dangerous if accessed concurrently.  I copied this from lab0.  Would it be okay to pass it in as a function parameter?

Reply:
@rokane I don't see how to do this without changing the function signature stipulated for this lab.  Is the alternative approach you are suggesting to add an extra parameter?

Reply:
Sorry I just realised the above comment might not have all context. "do this" 😆 .

The reason this declared outside of the `fib` function is so it can be modified to allow testing of the function print output.  A bit unconventional but this seems to be the approach recommended for purposes of this course.

Reply:
This is the accepted approach on several merged PRs.  I am going to resolve conversation for this issue as "Won't Fix".  So hopefully it can be merged as well 😄

Reply:
@rokane - necessary for this kind of lab setup (executable printing to stdout & 100% coverage)

Reply:
@nicholascross apologies on the late reply. I hadn't checked the other PR's yet, yours was my first. All good :)